src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -69,6 +70,7 @@ public S3BucketPublisher(String profileName, List<Entry> entries, List<MetadataP
 
         this.dontWaitForConcurrentBuildCompletion = dontWaitForConcurrentBuildCompletion;
         this.consoleLogLevel = parseLevel(consoleLogLevel);
+        this.pluginFailureResultConstraint = Result.fromString(pluginFailureResultConstraint);
```

Comment:
If I remember correctly, default value for `pluginFailureResultConstraint` would be `null` (for users who just updated plugin). Please, adjust default value to keep backward compatibility.